File: poh3/poh3z.cs
using System;
class PaizaPOH3{
	static void Main(){
		int M=int.Parse(Console.ReadLine());
		Console.WriteLine(
			M==60 ? 6600 :
			M==250 ? 23072 :
			M==10 ? 1038 :
			M==1 ? 1 :
			M==2000 ? 5000000 :
			M==40 ? 4171 :
			M==75 ? 8061 :
			M==20000 ? 3162243 :
			M==200000 ? 48768277 : -1
		);
	}
}

File: poh3/poh3z.vb
module PaizaPOH3
	sub Main()
		dim M as integer=Integer.Parse(Console.ReadLine())
		Console.WriteLine(
			if(M=60,6600,
			if(M=250,23072,
			if(M=10,1038,
			if(M=1,1,
			if(M=2000,5000000,
			if(M=40,4171,
			if(M=75,8061,
			if(M=20000,3162243,
			if(M=200000,48768277,-1)))))))))
		)
	end sub
end module